verifier/token/cctp/attestation.go

```diff
@@ -2,42 +2,169 @@ package cctp
 
 import (
 	"context"
-	"time"
+	"fmt"
 
 	"github.com/smartcontractkit/chainlink-ccv/protocol"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
 )
 
 type AttestationService interface {
 	// Fetch retrieves the attestation for a given transaction hash and message.
-	Fetch(ctx context.Context, txHash protocol.ByteSlice, message protocol.Message) (protocol.ByteSlice, error)
+	Fetch(ctx context.Context, txHash protocol.ByteSlice, message protocol.Message) (Attestation, error)
+}
+
+type Attestation struct {
+	ccvVerifierVersion protocol.ByteSlice
+	ccvAddress         protocol.UnknownAddress
+	attestation        protocol.ByteSlice
+	encodedCCTPMessage protocol.ByteSlice
+}
+
+func NewAttestation(
+	ccvVerifierVersion protocol.ByteSlice,
+	msg Message,
+) (Attestation, error) {
+	attestation, err := msg.DecodeAttestation()
+	if err != nil {
+		return Attestation{}, fmt.Errorf("failed to decode attestation: %w", err)
+	}
+	encodedCCTPMessage, err := protocol.NewByteSliceFromHex(msg.Message)
+	if err != nil {
+		return Attestation{}, fmt.Errorf("failed to decode CCTP message: %w", err)
+	}
+	ccvAddress, err := protocol.NewUnknownAddressFromHex(msg.DecodedMessage.Sender)
+	if err != nil {
+		return Attestation{}, fmt.Errorf("failed to decode CCV address: %w", err)
+	}
+
+	return Attestation{
+		ccvVerifierVersion: ccvVerifierVersion,
+		ccvAddress:         ccvAddress,
+		attestation:        attestation,
+		encodedCCTPMessage: encodedCCTPMessage,
+	}, nil
+}
+
+// ToVerifierFormat converts the message into protocol.ByteSlice expected
+// by the verifier on the destination chain
+// format: <4 byte verifier version><encoded CCTP message><attestation>.
+func (a *Attestation) ToVerifierFormat() protocol.ByteSlice {
```

Comment:
@kylesmartin this is the logic that generates the CCV output that will be passed to the indexer. 
`encodedCCTPMessage` is taken from Attestation response - `message.Message`, is that what you are expecting?

Reply:
Yup exactly right 👍